integration_tests/test_dict_increment.py

```diff
@@ -0,0 +1,55 @@
+from lpython import i32, f64
+
+def test_dict_increment():
+    d_int_int: dict[i32, i32]
+    d_int_float: dict[i32, f64]
+    d_str_int: dict[str, i32]
+    d_int_str: dict[i32, str]
+    i1: i32
+    i2: i32
+    j1: f64
+    s1: str
+
+    d_int_int = {0: 1}
+    d_int_int[0] += 1000
+    assert d_int_int[0] == 1001
+
+    i2 = 1
+    d_int_int = {1: i2}
+    for i1 in range(10):
+        d_int_int[1] += i1
```

Comment:
Also test with runtime keys. Right now all the keys in your test are compile time constants.

Reply:
Added